qiskit/providers/backend.py

```diff
@@ -283,7 +287,7 @@ class BackendV2(Backend, ABC):
     would contain an int for how many shots to execute.
 
     If migrating a provider from :class:`~qiskit.providers.BackendV1` or
-    :class:`~qiskit.providers.BaseBackend` one thing to keep in mind is for
+    :class:`~qiskit.providers.Backend` one thing to keep in mind is for
```

Comment:
We can just remove the `BaseBackend` here. Nothing really should directly be using `Backend` as it's just a common base class/type. This was just a leftover reference to BackendV0 what existed before we versioned this interface.

Reply:
I can remove it here since we're already touching it - there's still several references to `BaseBackend` and (I think) `Backend` throughout Qiskit that we'll need to squash as part of a more complete effort.

Reply:
Done in 7e5bf4b6c.